### ISSUE TITLE
Enhance/rearchitect Engine to allow mapping individual fail/reasons to different WCAG Success Criteria 

### ISSUE THREAD

**Issue by philljenkins** [IBMa/equal-access#1476]
## Detailed description on work being done to a rule or helping file: 
--> Rules today _via the ruleID_ map to one or more WCAG SCs. Each Rule has multiple fail id's/messages. Currently all messages (fail/reasons) in the ruleID **_all_** map to the same WCAG SC as the ruleID.  In other words, the ruleID dictates the mapping not the individual fail ids/reasons.

We would need to rearchitect the Engine's ruleID mapping to allow a single rule's fail ids messages to map to different SC's.

This would allow better alignment/mapping with ACT's rule mapping. For example, even though the Checker fails or passes an ACT test case, the Engine reports it against a different WCAG SC than what ACT reports it against. That's because the Checker's ruleID maps **_all_** the fail/messages by ruleID, not by fail/response message.

This should also allow the Engine to report some fails against a new and/or non-WCAG SC.  ACT reports some rules against a "spec failure" and not necessarily against a WCAG SC. WCAG 2.2 SCs are coming too.

The other options for now, without doing this work, would be to:
1. split the rule and fail/reasons into different ruleIDs that map to different WCAG SCs (if possible). 
2. live with the mismatch for now being reported as a "**_partial_**" matching of some rules against ACT
3. Or create a new SC ( a type of no-op/non acc issue) that isn't WCAG based.

This work assumes that the ACT mapping is correct. If not, then an issue is/will be opened against the ACT rule (See [About ACT Rules](https://wai-wcag-act-rules.netlify.app/standards-guidelines/act/rules/about/) for background)

### This ticket is for the following (select only one): 
- [x] ~~`New rule`~~
- [x] `Update to all current Checker rules`
- [x] ~~`Helping text for a rule`~~
- [x] ~~`other`~~

### Does this work has a dependency for a rule or help file, if yes please list the ticket(s):  
none

### Provide test file location: 

ACT reports the [Checker implementation status](https://wai-wcag-act-rules.netlify.app/standards-guidelines/act/implementations/equal-access/) after each Checker deployment, such as this screenshot dated 16 May 2023:
![image](https://media.github.ibm.com/user/37202/files/05da986f-0b7c-49b4-a4c6-8ddceb328819)
Note: the version no. "**_3.1.42-rc.0_**" reported in the screenshot is incorrect. It should be v3.1.48 for the results dated 16 May 2023. This inconsistency is a known bug with the ACT implementation report.

**Comment by philljenkins** [comment]
Also see ACT vs Checker mapping issue to WCAG SC's
#1469 
#1468 
and any possible issues in Epic #1463 ACT Issue Holder

**Comment by tombrunet** [comment]
@philljenkins The architecture work is already done. Are there separate issues for the ACT work? If so, can we close this?

**Comment by philljenkins** [comment]
@tombrunet I think we can close this issue. but I don't think any current "separate" issues are "up-to-date", so we may need a task to "gather" the missmatches
